src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,264 @@
+        this.handlers = handlers;
+        this.defaultName = defaultName;
+    }
+
+    private AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
+        this.customName = customName;
+        this.defaultName = defaultName;
+    }
+
+    /**
+     * Get the name of this object. For players this returns their username
+     */
+    @Override
+    public String getName()
+    {
+        return hasCustomName() ? customName : defaultName;
+    }
+
+    /**
+     * Method to set the name of this {@link IWorldNameable}
+     *
+     * @param name The new name of this {@link IWorldNameable}, or null to reset it to its default.
+     */
+    @Override
+    public void setName(@Nullable final String name)
+    {
+        this.customName = name;
+    }
+
+    /**
+     * Returns true if this thing is named
+     */
+    @Override
+    public boolean hasCustomName()
+    {
+        return customName != null;
+    }
+
+    /**
+     * Get the formatted ChatComponent that will be used for the sender's username in chat
+     */
+    @Override
+    public ITextComponent getDisplayName()
+    {
+        return new TextComponentString(getName());
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        final NBTTagCompound compound = new NBTTagCompound();
+
+        int index = 0;
+        final NBTTagList handlerList = new NBTTagList();
+        final NBTTagList indexList = new NBTTagList();
+        for(IItemHandlerModifiable handlerModifiable : handlers) {
+            if (handlerModifiable instanceof INBTSerializable)
+            {
+                INBTSerializable serializable = (INBTSerializable) handlerModifiable;
+                handlerList.appendTag(serializable.serializeNBT());
+                indexList.appendTag(new NBTTagInt(index++));
+            }
+        }
+
+        compound.setTag(NBT_KEY_HANDLERS, handlerList);
+
+        if (hasCustomName()) {
+            compound.setString(NBT_KEY_NAME, customName);
+        }
+
+        return compound;
+    }
+
+    @Override
+    public void deserializeNBT(final NBTTagCompound nbt)
+    {
+        final NBTTagList handlerList = nbt.getTagList(NBT_KEY_NAME, Constants.NBT.TAG_COMPOUND);
+        final NBTTagList indexList = nbt.getTagList(NBT_KEY_HANDLERS_INDEXLIST, Constants.NBT.TAG_INT);
+
+        if (handlerList.tagCount() == handlers.length){
+            for (int i = 0; i < handlerList.tagCount(); i++)
+            {
+                final NBTTagCompound handlerCompound = handlerList.getCompoundTagAt(i);
+                final IItemHandlerModifiable modifiable = handlers[indexList.getIntAt(i)];
+                if (modifiable instanceof INBTSerializable)
+                {
+                    INBTSerializable serializable = (INBTSerializable) modifiable;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'serializable' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)